keps/3122-expose-flavors-in-localqueue-status/README.md

```diff
@@ -0,0 +1,158 @@
+# KEP-3122: Expose Flavors in LocalQueue Status
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation overview](#implementation-overview)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a new status field in LocalQueue, allowing users to see
+all currently available ResourceFlavors in the LocalQueue.
+
+## Motivation
+
+Currently, users without RBAC access to ResourceFlavors cannot view the list
+of available flavors. Depending on the RBAC rules, users might also lack read
+access to ClusterQueues. Providing users with information about the available
+flavors is useful, as it gives them an idea of the capabilities provided by
+a LocalQueue (e.g., a flavor might include newer GPUs).
+
+### Goals
+
+- Provide a possibility to see all currently available ResourceFlavors in 
+  the LocalQueue.
+
+### Non-Goals
+
+- Verify that the ResourceFlavors exist and show only the existing flavors.
+
+## Proposal
+
+Introduce a new status field `availableFlavors` in LocalQueue 
+that will be updated when ClusterQueue flavors are modified.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user I want to see the list of all ResourceFlavors available in each LocalQueue 
+due to the RBAC configuration for ClusterQueue in my cluster I cannot inspect the 
+ClusterQueue objects directly, only LocalQueues.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### API
+
+Create `AvailableFlavor` API object:
+
+```go
+type AvailableFlavor struct {
+  // name of the flavor.
+  Name ResourceFlavorReference `json:"name"`
+  // resources used in the flavor.
+  // +listType=set
+  // +optional
+  Resources []string `json:"resources,omitempty"`
+}
+```
+
+Modify `LocalQueueStatus` API object:
+
+```go
+// LocalQueueStatus defines the observed state of LocalQueue
+type LocalQueueStatus struct {
+	...
+	// availableFlavors lists all currently available ResourceFlavors
+	// in specified ClusterQueue.
+	//
+	// +listType=map
+	// +listMapKey=name
+	AvailableFlavors []AvailableFlavor `json:"availableFlavors,omitempty"`
+}
+```
+
+### Implementation overview
```

Comment:
What happens if the tied ClusterQueue is forcefully removed?
Indeed, the clusterQueues are not removed by validations, but we can forcefully remove the ClusterQueue by ignoring validations.

Reply:
We are still checking the cache after removing the ClusterQueue and updating the status of the LocalQueue if it has changed. So, it should work. I also checked this manually, and the status was updated.

Reply:
I see. So, it would be great to add testing in the implementation PR.

Reply:
Done